### ISSUE TITLE
Controllers unavailable on Pico 3 (accepting Terms of Service)

### ISSUE THREAD

**Issue by ipeakermit** [Igalia/wolvic#407]
## Configuration

Wolvic version: commit 9d06ab663d60e5ef110f7598dc1a01c7cae330bf
Wolvic build ID: 

Hardware: Pico Neo 3 Pro (4.8.19, 2022.12.22)

## Steps to Reproduce

1. ./gradlew clean
2. ./gradlew installPicoxrArm64WorldDebug
3. Open Wolvic on Pico Neo 3

## Current Behavior

Wolvic splash screen appears showing Terms of Service dialog box (requires selecting "Agree" to proceed), but controllers are invisible/disabled

![Screenshot from 2023-01-11 10-55-39](https://user-images.githubusercontent.com/11951395/211688768-622d769a-2f29-4384-9309-66a5c07825ce.png)

## Expected Behavior

Controller should be enabled in order to click "Agree"

## Possible Solution

Seems to be related to hand tracking initialization

## Error Logs and Stack Traces

<details open>

```
```

</details>

**Comment by svillar** [comment]
We do only support Pico4 in the OpenXR backend for the time being, that's why you are not seeing the controllers. We could think about adding support for other devices if there is a reliable way to recognize devices just using OpenXR.

**Comment by ipeakermit** [comment]
Seems to be related to hand tracking initialization. Commenting out the code under openxr/cpp/OpenXRInputSource.cpp, OpenXRInputSource::Initialize, "Initialize hand tracking, if supported" reinstates controller models, and I was able to succesfully run a Youtube 360 demo. Will try and turn this into a PR

**Comment by svillar** [comment]
> Seems to be related to hand tracking initialization. Commenting out the code under openxr/cpp/OpenXRInputSource.cpp, OpenXRInputSource::Initialize, "Initialize hand tracking, if supported" reinstates controller models, and I was able to succesfully run a Youtube 360 demo. Will try and turn this into a PR

That's awesome. Looking forward to reviewing it!

**Comment by ipeakermit** [comment]
Is there some further info on how to build a compliant wolvic with WebXR? The README.md specifies a local build of https://github.com/Igalia/gecko-dev/ on the FIREFOX_103_0_2_RELEASE branch. After muddling through the use of mach, my resulting Pico 3 wolvic build crashes immediately with: "java.lang.RuntimeException: Could not find mozglue path". I am not at all sure that I built gecko correctly.

**Comment by svillar** [comment]
> Is there some further info on how to build a compliant wolvic with WebXR? The README.md specifies a local build of https://github.com/Igalia/gecko-dev/ on the FIREFOX_103_0_2_RELEASE branch. After muddling through the use of mach, my resulting Pico 3 wolvic build crashes immediately with: "java.lang.RuntimeException: Could not find mozglue path". I am not at all sure that I built gecko correctly.

How does your `mozconfig` look like? You should have these options at least
```
ac_add_options --with-java-bin-path=/usr/lib/jvm/java-11-openjdk-amd64/bin
ac_add_options --enable-application=mobile/android
ac_add_options --target=aarch64
ac_add_options --enable-optimize --disable-debug --enable-geckoview-lite
```

Apart from that it's critical to set the proper paths in `local.properties`
```
dependencySubstitutions.geckoviewTopsrcdir=PATH_TO_GECKO_DEV
dependencySubstitutions.geckoviewTopobjdir=PATH_TO_GECKO_DEV/obj-aarch64-unknown-linux-android
```

**Comment by ipeakermit** [comment]
> > Is there some further info on how to build a compliant wolvic with WebXR? The README.md specifies a local build of https://github.com/Igalia/gecko-dev/ on the FIREFOX_103_0_2_RELEASE branch. After muddling through the use of mach, my resulting Pico 3 wolvic build crashes immediately with: "java.lang.RuntimeException: Could not find mozglue path". I am not at all sure that I built gecko correctly.
> 
> How does your `mozconfig` look like? You should have these options at least
> 
> ```
> ac_add_options --with-java-bin-path=/usr/lib/jvm/java-11-openjdk-amd64/bin
> ac_add_options --enable-application=mobile/android
> ac_add_options --target=aarch64
> ac_add_options --enable-optimize --disable-debug --enable-geckoview-lite
> ```
> 
> Apart from that it's critical to set the proper paths in `local.properties`
> 
> ```
> dependencySubstitutions.geckoviewTopsrcdir=PATH_TO_GECKO_DEV
> dependencySubstitutions.geckoviewTopobjdir=PATH_TO_GECKO_DEV/obj-aarch64-unknown-linux-android
> ```

Thanks. Sorry, I also asked the same question here (with similar response fixing the problem) https://github.com/Igalia/wolvic/issues/416

**Comment by ipeakermit** [comment]
Thanks @svillar . I now have Wolvic working on the Pico 3 and running WebXR applications like Hubs and Moonrider.xyz based on the discussion above. Patch should be fairly easy, but note that initially this will just be a commenting out of the hand tracking initialization.

**Comment by neur0ns** [comment]
The same on Pico 4 (model A8250; PicoOS - 5.0.3; Android sdk 29). 

But, the UI is 3-4 meters lower than my horizon.

**Comment by svillar** [comment]
> The same on Pico 4 (model A8250; PicoOS - 5.0.3; Android sdk 29).
> 
> But, the UI is 3-4 meters lower than my horizon.

Which Wolvic version are you using? Please try with 1.3 or 1.3.1

**Comment by neur0ns** [comment]
Resolved by update to 5.2.7 of PicoOS.

P.s. I tried the latest version.

**Comment by svillar** [comment]
Closing then...
